File: adaptors/php/basic_commands_example.php
<?php

require_once __DIR__ . '/JsonStorageConnectionClass.php';

echo "Starting basic commands example...\n";

try {
    // Initialize connection
    $storage = new JsonStorageConnection('localhost', 5555);
    echo "Connecting to server...\n";
    $storage->connect();
    
    // Basic commands demonstration
    echo "\nRunning basic commands:\n";
    
    // PING test
    echo "\n=== PING Test ===\n";
    $response = $storage->get("PING");
    echo "PING response: " . $response . "\n";
    
    // SET command
    echo "\n=== SET Command ===\n";
    $key = "example:test:1";
    $value = "Hello from basic commands!";
    $success = $storage->set($key, $value);
    echo "SET result: " . ($success ? "SUCCESS" : "FAILED") . "\n";
    
    // GET command
    echo "\n=== GET Command ===\n";
    $retrieved = $storage->get($key);
    echo "GET result: " . $retrieved . "\n";
    
    // SET with TTL
    echo "\n=== SET with TTL ===\n";
    $ttlKey = "example:ttl:1";
    $ttlValue = "This will expire in 5 seconds";
    $storage->set($ttlKey, $ttlValue, 5);
    echo "Initial TTL value: " . $storage->get($ttlKey) . "\n";
    echo "TTL remaining: " . $storage->ttl($ttlKey) . " seconds\n";
    
    // Wait and check TTL
    sleep(2);
    echo "After 2 seconds - TTL remaining: " . $storage->ttl($ttlKey) . " seconds\n";
    
    // DELETE command
    echo "\n=== DELETE Command ===\n";
    $storage->del($key);
    $checkDeleted = $storage->get($key);
    echo "After DELETE - value exists?: " . ($checkDeleted === null ? "No" : "Yes") . "\n";
    
} catch (Exception $e) {
    echo "\nError: " . $e->getMessage() . "\n";
    echo "Stack trace:\n" . $e->getTraceAsString() . "\n";
} finally {
    if (isset($storage)) {
        $storage->close();
        echo "\nConnection closed.\n";
    }
}

File: adaptors/python/basic_commands_example.py
#!/usr/bin/env python3

from json_storage_connection import JsonStorageConnection
import time

print("Starting basic commands example...\n")

try:
    # Initialize connection
    storage = JsonStorageConnection('localhost', 5555)
    print("Connecting to server...")
    storage.connect()
    
    # Basic commands demonstration
    print("\nRunning basic commands:")
    
    # PING test
    print("\n=== PING Test ===")
    response = storage.get("PING")
    print(f"PING response: {response}")
    
    # SET command
    print("\n=== SET Command ===")
    key = "example:test:1"
    value = "Hello from basic commands!"
    success = storage.set(key, value)
    print(f"SET result: {'SUCCESS' if success else 'FAILED'}")
    
    # GET command
    print("\n=== GET Command ===")
    retrieved = storage.get(key)
    print(f"GET result: {retrieved}")
    
    # SET with TTL
    print("\n=== SET with TTL ===")
    ttl_key = "example:ttl:1"
    ttl_value = "This will expire in 5 seconds"
    storage.set(ttl_key, ttl_value, 5)
    print(f"Initial TTL value: {storage.get(ttl_key)}")
    print(f"TTL remaining: {storage.ttl(ttl_key)} seconds")
    
    # Wait and check TTL
    time.sleep(2)
    print(f"After 2 seconds - TTL remaining: {storage.ttl(ttl_key)} seconds")
    
    # DELETE command
    print("\n=== DELETE Command ===")
    storage.delete(key)
    check_deleted = storage.get(key)
    print(f"After DELETE - value exists?: {'Yes' if check_deleted is not None else 'No'}")

except Exception as e:
    print(f"\nError: {str(e)}")
    import traceback
    print("Stack trace:")
    print(traceback.format_exc())
finally:
    if 'storage' in locals():
        storage.close()
        print("\nConnection closed.")